containers/src/Kokkos_DynRankView.hpp

```diff
@@ -875,47 +954,70 @@ class DynRankView : private View<DataType*******, Properties...> {
+#ifndef KOKKOS_ENABLE_IMPL_VIEW_LEGACY
+      : view_type(attach_accessor_arg_if_needed(arg_prop, arg_layout),
+                  drdtraits::template createLayout<traits, P...>(arg_prop,
+                                                                 arg_layout)),
+        m_rank(drdtraits::computeRank(arg_prop, arg_layout) -
+               (traits::impl_is_customized &&
+                        !Kokkos::Impl::ViewCtorProp<P...>::has_accessor_arg
+                    ? 1
+                    : 0)){}
+#else
       : view_type(arg_prop, drdtraits::template createLayout<traits, P...>(
                                 arg_prop, arg_layout)),
-        m_rank(drdtraits::computeRank(arg_prop, arg_layout)) {}
+        m_rank(drdtraits::computeRank(arg_prop, arg_layout)) {
+  }
+#endif
+
 #ifdef KOKKOS_IMPL_SKIP_OPTIMIZATION
 #pragma GCC pop_options
 #pragma pop
 #undef KOKKOS_IMPL_SKIP_OPTIMIZATION
 #endif
 
-  //----------------------------------------
-  // Constructor(s)
-
-  // Simple dimension-only layout
-  // We need two variants to avoid calling host function from host device
-  // function warnings
-  template <class... P>
-  explicit KOKKOS_FUNCTION DynRankView(
-      const Kokkos::Impl::ViewCtorProp<P...>& arg_prop,
-      std::enable_if_t<Kokkos::Impl::ViewCtorProp<P...>::has_pointer,
-                       const size_t>
-          arg_N0          = KOKKOS_INVALID_INDEX,
-      const size_t arg_N1 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N2 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N3 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N4 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N5 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N6 = KOKKOS_INVALID_INDEX,
-      const size_t arg_N7 = KOKKOS_INVALID_INDEX)
+        //----------------------------------------
+        // Constructor(s)
+
+        // Simple dimension-only layout
+        // We need two variants to avoid calling host function from host device
+        // function warnings
+        template <class... P>
+        explicit KOKKOS_FUNCTION DynRankView(
+            const Kokkos::Impl::ViewCtorProp<P...>& arg_prop,
+            std::enable_if_t<Kokkos::Impl::ViewCtorProp<P...>::has_pointer,
+                             const size_t>
+                arg_N0          = KOKKOS_INVALID_INDEX,
+            const size_t arg_N1 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N2 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N3 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N4 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N5 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N6 = KOKKOS_INVALID_INDEX,
+            const size_t arg_N7 = KOKKOS_INVALID_INDEX)
       : DynRankView(arg_prop, typename traits::array_layout(
                                   arg_N0, arg_N1, arg_N2, arg_N3, arg_N4,
-                                  arg_N5, arg_N6, arg_N7)) {}
+                                  arg_N5, arg_N6, arg_N7)) {
+  }
```

Comment:
Any idea what happened here?

Reply:
ifdefs in the previous ctors are confusing clang tidy.